spinalcordtoolbox/scripts/sct_compute_hausdorff_distance.py

```diff
@@ -274,7 +274,8 @@ def compute_dist_2im_2d(self):
         nx1, ny1, nz1, nt1, px1, py1, pz1, pt1 = self.im1.dim
         nx2, ny2, nz2, nt2, px2, py2, pz2, pt2 = self.im2.dim
 
-        assert np.isclose(px1, px2) and np.isclose(py1, py2) and np.isclose(px1, py1)
+        if not (np.isclose(px1, px2) and np.isclose(py1, py2) and np.isclose(px1, py1)):
+            raise ValueError("Resolutions are not close enough")
```

Comment:
This is a bit of a weird one. It's checking the resolutions of two 2D images by comparing the X resolutions, the Y resolutions, but also the X1 and Y1 resolutions?

In other words, the images need to be A) isotropic and B) have similar resolutions? Given that there are two different conditions, maybe we should have two different error messages to disambiguate the error cases:

```suggestion
        if not all(np.isclose(px1, py1) and np.isclose(px2, py2)):
            raise ValueError(f"2D resolutions must both be isotropic, but got '{(px1, py1)}' and '{(px2, py2)}' instead.")
        if not np.all(np.isclose(px1, px2) and np.isclose(py1, py2)):
            raise ValueError(f"2D resolutions must be close to one another, but got '{(px1, py1)}' and '{(px2, py2)}' instead.")
```

Reply:
Technically there is a world where A == B == C == D but A != D, but that doesn't feel intentional on the part of the old check, given the symmetry in the logic.

Reply:
That's a much better set of checks, thank you for digging into it! I had noticed that there was a non-symmetric cross-comparison and thought it was curious, but decided to just go for the safe mechanical change. Your explanation makes a lot of sense!